File: CS315-HW1/Codes/altiparmak_gokberk.lua
local capitals = {}
capitals["Turkey"] = "Istanbul"
capitals["Germany"] = "Berlin"
for k, v in pairs(capitals) do print(k, v) end
print("\n")
print(capitals["Turkey"])
capitals["Italy"] = "Rome"
print("\n")
for k,v in pairs(capitals) do print(k, v) end
print("\n")
capitals.Germany = nil
for k,v in pairs(capitals) do print(k, v) end
capitals["Turkey"] = "Ankara"
print("\n")
for k,v in pairs(capitals) do print(k, v) end
print("\n")
if capitals["Turkey"] ~= nil then print("true")
else print("false") end

File: CS315-HW1/Codes/altiparmak_gokberk.rs
use std::collections::HashMap;

fn foo(map: &mut HashMap<&str,&str>) ->(){
  for (key, value) in map.iter() {
    println!("{} - {}", key, value);
  }
  println!("------");
}

fn main() {
  let mut capitals = HashMap::new();
  capitals.insert("Turkey", "Istanbul");
  capitals.insert("Germany", "Berlin");
  foo(&mut capitals);
  println!("{:?}", capitals["Turkey"]);
  println!("------");
  capitals.insert("Italy", "Rome");
  foo(&mut capitals);
  capitals.remove("Germany");
  foo(&mut capitals);
  *capitals.get_mut("Turkey").unwrap() = "Ankara";
  foo(&mut capitals);
  println!("{:?}", capitals.contains_key("Turkey"));
  println!("{:?}", capitals.values().any(|&x| x == "Ankara"));
  println!("------");
  foo(&mut capitals);
}